rotkehlchen/db/filtering.py

```diff
@@ -1575,6 +1583,31 @@ def make(
         return filter_query
 
 
+class LocationAssetMappingsFilterQuery(DBFilterQuery):
+    location_filter: DBNullableLocationFilter | None = None
```

Comment:
This has some special logic regarding the location argument. Can you write a docstring explaining how it works?